### PR TITLE
fix: delete non-img file btn and workflow

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
### Summary

- Fixed the delete button for non-image files
- Simplified the workflow for deletion (migrating from jQuery)

### Will affect visual aspect of the product

YES

### Screenshots 

https://github.com/user-attachments/assets/3837df43-7eb5-4c4f-a447-2e69b77b2eed

### Test instructions

- Use a File Input Field
- Add an image and non-img file (like zip)
- Check the delete button.

## Check before Pull Request is ready:

* [ ] I have [written a test](CONTRIBUTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](CONTRIBUTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](CONTRIBUTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](CONTRIBUTING.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](CONTRIBUTING.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

Closes https://github.com/Codeinwp/ppom-pro/issues/422

### CODE REVIEW

File: js/file-upload.js
Comment (HardeepAsrani) [inline js/file-upload.js:93]: We can confirm to make sure the selector exists before adding an event listener.

File: inc/files.php
Comment (HardeepAsrani) [inline inc/files.php:497]: We can remove if it's not used.

File: inc/files.php
Comment (HardeepAsrani) [inline inc/files.php:501]: If we can do this without affecting the styles, it's always good to use `<button>` instead of anchor with empty href. It's also better for people using screen readers. As a rule of thumb, use the button if the action happens on the same page. If the action takes you to a different page, you can use an anchor.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
Looks good overall, left a couple of small changes.

**Comment by AndreeaCristinaRadacina** [comment]
It works as expected, @Soare-Robert-Daniel! Thank you for taking care.

I'll wait for the other issues before moving it to Ready to merge.

File: inc/files.php
Comment (carlalexander) [inline inc/files.php:351]: Great addition 🥳

**Review by carlalexander — APPROVED** [review]
Nothing to add beyond what @HardeepAsrani already highlighted. Once he's ok with it, we're good to go! 🚀

File: inc/files.php
Comment (HardeepAsrani) [inline inc/files.php:483]: You missed the href tag in the button. 😄

Comment (HardeepAsrani) [reply inc/files.php:483]: Also I see there's a closing `</span>` without an opening. I think in past it was for icons but at some point we removed it but forgot to remove the closing tag.

File: inc/files.php
Comment (HardeepAsrani) [inline inc/files.php:499]: Same here with the href.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
Thanks for the changes, just a couple of smaller things left.

**Review by HardeepAsrani — APPROVED** [review]

**Pull request merged**
